internal/payloads/create_source_payload.go

```diff
@@ -0,0 +1,62 @@
+package payloads
+
+import (
+	"fmt"
+	"strings"
+)
+
+type CreateSourceResponse struct {
+	Applications []struct {
+		ApplicationTypeID string `json:"application_type_id"`
+		CreatedAt         string `json:"created_at"`
+		ID                string `json:"id"`
+		SourceID          string `json:"source_id"`
+		UpdatedAt         string `json:"updated_at"`
+	} `json:"applications"`
+	Authentications []struct {
+		Authtype     string `json:"authtype"`
+		ID           string `json:"id"`
+		ResourceID   string `json:"resource_id"`
+		ResourceType string `json:"resource_type"`
+		Username     string `json:"username"`
+	} `json:"authentications"`
+	Endpoints []interface{} `json:"endpoints"`
+	Sources   []struct {
+		AppCreationWorkflow string `json:"app_creation_workflow"`
+		CreatedAt           string `json:"created_at"`
+		ID                  string `json:"id"`
+		Name                string `json:"name"`
+		SourceTypeID        string `json:"source_type_id"`
+		UID                 string `json:"uid"`
+		UpdatedAt           string `json:"updated_at"`
+	} `json:"sources"`
+}
+
+func CreateSourceRequest(sourceName string, ARN string) *strings.Reader {
+	reqStr := fmt.Sprintf(`{
+		"sources": [
+				{
+						"name": "%s",
+						"source_type_name": "amazon",
+						"app_creation_workflow": "manual_configuration"
+				}
+		],
+		"applications": [
+				{
+						"source_name": "%s",
+						"application_type_name": "provisioning"
+				}
+		],
+		"authentications": [
+				{
+						"resource_type": "Application",
+						"resource_name": "provisioning",
+						"username": "%s",
+						"authtype":"provisioning-arn"
+				}
+		]
+	}`, sourceName, sourceName, ARN)
+
+	body := strings.NewReader(reqStr)
+	return body
```

Comment:
Wow, okay okay. I have no problems with that, this is also fast :) Just make sure arguments are JSON-escaped.